tools/filesystem/file_test.go

```diff
@@ -43,6 +43,33 @@ func TestNewFileFromFromPath(t *testing.T) {
 	}
 }
 
+func TestNewFileFromBytes(t *testing.T) {
+	testDir := createTestDir(t)
+	defer os.RemoveAll(testDir)
+
+	// nil bytes
+	if _, err := filesystem.NewFileFromBytes(nil, "photo.jpg"); err == nil {
+		t.Fatal("Expected error, got nil")
+	}
+
+	// zero bytes
+	if _, err := filesystem.NewFileFromBytes([]byte{}, "photo.jpg"); err == nil {
+		t.Fatal("Expected error, got nil")
+	}
+
+	originalName := "notes.txt"
+	f, err := filesystem.NewFileFromBytes([]byte("TODO: Write Tests\n"), originalName)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if f.Size != 18 {
+		t.Fatalf("Expected Size %v, got %v", 18, f.Size)
+	}
+	if f.OriginalName != originalName {
+		t.Fatalf("Expected originalName %q, got %q", originalName, f.OriginalName)
+	}
```

Comment:
Can you also add a test for the "normalized" name?
You can use "originalName" with some strange characters and then check it against a regex pattern (we do something similar in the existing tests).